charts/cryostat/templates/cryostat_deployment.yaml

```diff
@@ -6,13 +6,15 @@ metadata:
   name: {{ include "cryostat.fullname" . }}
   labels:
     {{- include "cryostat.labels" . | nindent 4 }}
+    app.kubernetes.io/component: cryostat
 spec:
   replicas: 1
   strategy:
     type: Recreate
   selector:
     matchLabels:
       {{- include "cryostat.selectorLabels" . | nindent 6 }}
+      app.kubernetes.io/component: cryostat
```

Comment:
Related discussion: https://github.com/helm/helm/issues/7082

Reply:
If I understand correctly, this would just be a one-time requirement for users when upgrading from Cryostat 3.0 to 4.0, which I think is fine. Once on 4.0 it should be possible to upgrade without `--force` to ex. 4.1 so long as we don't make another similar breaking change.

Reply:
```bash
$ helm search repo cryostat
NAME               	CHART VERSION	APP VERSION	DESCRIPTION                                       
cryostatio/cryostat	1.0.1        	3.0.1      	Securely manage JFR recordings for your contain...

# Install 1.0.1 version
$ helm install cryostat-sample cryostatio/cryostat

# on this PR, upgrade 2.0.0-dev version
$ helm upgrade cryostat-sample charts/cryostat/
```

Reply:
This is just another temporary/one-time workaround the same as `--force` IMO, because we would have to do another rename again in the future if we change the labels again or do anything else similar.

Maybe a naming scheme like `{{ include "cryostat.fullname" . }}-v{{ cryostat.semver.major }}`, which would expand out to ex. `cryostat-v3`, could be usable across the various resources. This way their names stay stable within a major release version, but on major version upgrade when things are expected to change in a breaking way anyway, then the resources are renamed and upgrades can be done without force.

Reply:
Oh very nice! This naming scheme would be a much better approach ^^

Reply:
I guess this would be some work on another follow-up PR then.

Reply:
#200